staging/src/k8s.io/cri-api/pkg/apis/runtime/v1alpha2/api.proto

```diff
@@ -543,6 +543,21 @@ message LinuxContainerResources {
     string cpuset_cpus = 6;
     // CpusetMems constrains the allowed set of memory nodes. Default: "" (not specified).
     string cpuset_mems = 7;
+    // List of HugepageLimits. Default: nil (not specified).
+    // Container can have hugepage limits per page size.
+    // Each limit will be set on hugetlb.<size><unit-prefix>B.limit_in_bytes.
+    // For example, `pageSize=1GB`, `limit=1073741824` means setting `1073741824` bytes
+    // on hugetlb.1GB.limit_in_bytes of container level cgroup.
+    repeated HugepageLimit hugepage_limits = 8;
+}
+
+
+// HugepageLimit contains hugepages information for a container.
+message HugepageLimit {
+    // hugepage size in the format <size><unit-prefix>B (2MB, 1GB)
+    string page_size = 1;
```

Comment:
should it be `uint` that can be matched with corresponding entry in `/sys/devices/system/node/node*/hugepages/hugepages-*kB`  ?

Reply:
`Or idea to validate it always with what is available in /sys/fs/cgroup/hugetlb/hugetlb.* and fail if something doesn't match?`
=> Yes, it is my original thought.
I referred OCI [runtime spec](https://github.com/opencontainers/runtime-spec/blob/19e92ca817772b4466f2ed2b8d808dfb7a8ab4be/specs-go/config.go#L220).
Shoud i update it to `<unit-prefix-in-capital-letter>` to avoid ambiguity?
```
// LinuxHugepageLimit structure corresponds to limiting kernel hugepages
type LinuxHugepageLimit struct {
	// Pagesize is the hugepage size
	// Format: "<size><unit-prefix>B' (e.g. 64KB, 2MB, 1GB, etc.)
	Pagesize string `json:"pageSize"`
	// Limit is the limit of "hugepagesize" hugetlb usage
	Limit uint64 `json:"limit"`
}
```

Reply:
I think it is ok, but updating comment referencing OCI spec https://github.com/opencontainers/runtime-spec/blob/master/config-linux.md#huge-page-limits can be also a good idea. OCI spec is explicit here to define what this string should contain.

Reply:
I think updating the comment is nice too.

I agree with @kad that using an uint would be nice, but I still thing following the OSI spec is the best solution, so I prefer to keep it as a string. The size prefixes originate from here btw: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/mm/hugetlb_cgroup.c?id=e3280b54afed870d531571212f1fc375df39b7d2#n342